Applications/Spire/Source/TimeAndSales/TimeAndSalesTableView.cpp

```diff
@@ -131,6 +132,13 @@ void TimeAndSalesTableView::set_properties(
   }
 }
 
+bool TimeAndSalesTableView::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == m_table && event->type() == QEvent::Paint) {
+    m_header->viewport()->update();
```

Comment:
This might work but I'd prefer to avoid using event filters as much as possible. When does a table repaint itself? Even better would be to determine when does a table repainting require the header's viewport to be updated. Is it really on every single table paint event or is there perhaps another causal relationship you can make use of that doesn't require an event filter, for example a when a row is added, removed, or the table is scrolled.